### DISCUSSION TITLE
Enrolling more `aio-libs` projects to Tidelift

### DISCUSSION THREAD

**Discussion by webknjaz** [aio-libs/.github#33] [Announcements]
So Tidelift is a project that lets us collect some $$$ from the commercial users of open source. They try to track and distribute some funds based on what their subscribers use.
The idea is that maintainers subscribe and perform tasks that show up in their dashboard and for that, there's some monthly compensation. This is tied to individuals who are called "lifters". If several maintainers sign up, the funds can be distributed among them (can be equal or any other percentage-based agreement). Each lifter has to sign some agreement to join and once eligible, they use a Hyperwallet service to forward money to your bank account or PayPal. The tasks are mostly related to verifying that the metadata is correct (licenses, versioning) and confirming things like CVEs. They can also help with coordinating vulnerability reporting and disclosures (this is opt-in).

I submitted a few of the aio-libs projects there but it'd be great if there were more lifters, from those who maintain said projects more actively/regularly. Y'all know that I mostly do organizational stuff / docs / packaging / CI maintenance these days, but I'd also like to enable people who do other types of maintenance work get some of those donations too.

I really hope more of you could join me in this endeavor! AMA — I've been lifting many other projects for the past few years.

cc @achimnol @bdraco @Dreamsorcerer @hellysmile @jettify @mjpieters @Nothing4You @pohmelie @samuelcolvin @aio-libs/admins

P.S. Let me know which projects you want to join on Tidelift — they usually coordinate talking to maintainers through email threads so I'll need to tell them about you or confirm otherwise.

----

Some packages I applied to lift (the marked ones do receive income):
- [x] 💰 `aiohappyeyeballs`
- [x] 💰 `aiohttp`
- [x] 💰 `async-lru`
- [x] 💰 `frozenlist`
- [x] 💰 `idna_ssl`
- [x] 💰 `multidict`
- [x] 💰 `propcache`
- [x] 💰 `pytest-aiohttp`
- [x] 💰 `yarl`
- [x] 💰 `aiodns`
- [x] `aiocassandra`
- [x] `aioelasticsearch`
- [x] `aioftp`
- [x] `aioga`
- [x] `aiohttp-devtools`
- [x] `aiohttp-jinja2`
- [x] `aiohttp-session`
- [x] `aiojobs`
- [x] `aiomonitor`
- [x] `aiomysql`

**Comment by pohmelie** [comment]
The only project (from list above) I have strong expertise in is `aioftp`, so, if someday it will be «checked», then I can try this role.

**Reply by webknjaz** [reply]
It'll be checked upon approval by Tidelift, but it seems it doesn't generate money as of today. Still, it's possible to apply now so it's listed in your dashboard. I only marked things they approved earlier.

**Comment by webknjaz** [comment]
UPD: Tidelift folks approved the projects. I marked the ones that generate income currently, in the list above.
Again, if anybody wants to join — make sure to let me know so you could be added on Tidelift.

**Reply by Dreamsorcerer** [reply]
Just getting back round to this. Feel free to add me to any projects (particularly the aiohttp ones).

**Reply by webknjaz** [reply]
@Dreamsorcerer have you registered on Tidelift? They don't exactly have a button to add people skipping the bureaucracy. Try using this search box https://tidelift.com/about/lifter for every project (use the PyPI names!), navigate to pages like https://tidelift.com/lifter/search/pypi/aiohttp, scroll down (on some, the button is at the bottom, I think) and click `Lift your package`. Let me know if that works!
The next step would be for @katzj or his colleagues will email me+you to verify that you should indeed be added and ask about the money split that they enter into the system on their side somehow. They'll also want you to sign the docs before approving IIRC, and enroll you into Hyperwallet that they use for payouts.

P.S. I recommend applying to lift all the projects that you own outside @aio-libs as well. This way, if they start generating income in the future, you won't miss out.
P.P.S. If you want to discuss this more in private — find me on Matrix or drop me an email.

**Reply by Dreamsorcerer** [reply]
OK, they don't even have a register option, it just says to email them. Will try that now.

**Reply by webknjaz** [reply]
@Dreamsorcerer you can log in via GitHub/Google here https://tidelift.com/login (I haven't tried email+password so I don't know if entering it will suggest registering)

**Reply by Dreamsorcerer** [reply]
That's what told me to email support. For anyone else trying to sign up, you need to go to a package (https://tidelift.com/lifter/search/pypi/aiohttp) and then click the "lift your package" button at the bottom of the screen. That appears to be the only way to get to the register option.

**Reply by webknjaz** [reply]
> That appears to be the only way to get to the register option.

There's a login button in the header bar that takes you there:

![Screenshot_2024-02-29-15-39-11-68_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://github.com/aio-libs/.github/assets/578543/93305845-3ad4-4f19-a146-458386a5873e)

**Reply by Dreamsorcerer** [reply]
No, that's what I'm saying, it's not a register form. You try and login with Github and it tells you that you don't have an account and to email support. The only way to register is via the button on a package.

**Reply by Dreamsorcerer** [reply]
The register form looks pretty much identical, so you wouldn't notice anything different, except that it actually works when you login with Github.

**Reply by webknjaz** [reply]
Oh, interesting.. I didn't realize. I was so long ago when I registered.

**Reply by Dreamsorcerer** [reply]
Also a bit odd that there can be income from idna_ssl, given that the project was archived 3.5 years ago.

**Reply by webknjaz** [reply]
This means that it's still in people's dependency trees. It's not a big deal for us, I think.

**Reply by webknjaz** [reply]
@Dreamsorcerer I believe you're getting the same task reminders as me. Could you mark that you have 2FA on PyPI in the Tidelift interface so that they stop?

**Reply by Dreamsorcerer** [reply]
Yep, still catching up on tasks (haven't completed the tax stuff yet).

**Reply by webknjaz** [reply]
@Dreamsorcerer okay. I just wanted to make sure since I can't mark certain things as complete for other people.

I got my monthly receipt today so I figured I'd check in to see if there's anything left to do or any help needed.. All's good?

**Reply by Dreamsorcerer** [reply]
Yep, I got the same. Just need to find a printer/scanner to complete this tax form before I can withdraw it, seems nobody owns one these days...

**Reply by webknjaz** [reply]
Personally, I started using arbitrary electronic signing software. The first one found online allows me to draw a signature and put it onto a PDF which I than send. But if you just have a printer and no scanner, using mobile apps like Google Drive can do reasonable scans and outputs PDFs.

I know that apple devices (iPad?) have PDF apps where you can save a signature and store it for future use, putting it on any PDF whenever.

**Reply by Dreamsorcerer** [reply]
I already tried submitting a digital one (you can draw directly in Firefox/pdf.js these days), apparently you need the signature to be timestamped with an approved timestamp server, which they charge money for. I literally don't understand what the purpose is either, you can easily just submit the same signature to be timestamped again at a later date so it doesn't prove the signature didn't exist earlier, and they only accept documents dated in the last 5 days so there's no need to prove that it wasn't created at a later date. Completely pointless, and done in a way that has multiple points of possible corruption (compared with the free opentimestamp.org which is completely trustless).

**Reply by webknjaz** [reply]
Oh, I didn't even do a "digital signature" thing (in the asymmetric crypto sense). So it's just a picture, which would be equivalent to printing, drawing a signature with a pen and scanning that.

**Reply by Dreamsorcerer** [reply]
Yeah, I guess the question is how realistic it looks in order for them to not notice. Found a printer I can use at the weekend, so I'll just do "correctly".
